wordpress-mobile/WordPress-Android
# Media upload error codes wpcom jp in media lib

This PR is part of a series related to #12325 

The changes here should address the Media Library with WP.com/JP scenario. The equivalent scenarios for GB Editor will be covered in subsequent PRs.

This has a companion FluxC PR [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1657) 

Here below the error messages you should get from endpoint API for `rest_upload_limited_space`, `rest_upload_file_too_big` and `rest_upload_user_quota_exceeded`

```
rest_upload_limited_space|Not enough space to upload. 3,145 KB needed. Back

rest_upload_file_too_big|This file is too big. Files must be less than 1500000 KB in size. Back

rest_upload_user_quota_exceeded|You have used your space quota. Please delete files before uploading. Back
```


Note that we changed the FluxC to get the user message part of the string (that after the `|` sign) to be localized from the API (adding the `locale` query param that was not provided before).

One thing we should consider is maybe to filter out the ` Back` ending part, it does not seem to be a message to the user and not sure how it is used.

### To test:
The following apply both to WP.com and JP sites.

- Open the Media Library screen and try to upload a file
- Force trigger the 3 errors as described above (I can provide access to my sandbox or you can use Charles proxy rewrite feature or AS debugger for this)
- Check you get the API user message in the snackbar
- Check you get the message localized if you change app language

UPDATE: while further checking this I see the the message is cut in the snackbar since too long, maybe something we can discuss and address in another PR.

PR submission checklist:
